src/architect/project/project.utils.ts

```diff
@@ -0,0 +1,215 @@
+      }
+      const key = selection.name.toLowerCase();
+      const component_path = path.join(project_name, key, 'architect.yml');
+      const component_config = buildSpecFromPath(component_path);
+      console.log(`Linking ${key} with architect cli so it can be used as a dependency.`);
+      console.log(`% architect link ${component_path}`);
+      app.linkComponentPath(component_config.name, component_path);
+    }
+    const root_path = path.join(project_name, this.getRootComponent(selections), 'architect.yml');
+    console.log(`\nTry running your project now: \`architect dev ${root_path}\`\n`);
+  }
+
+  static async prompt(choices: any[], message: string): Promise<any> {
+    inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+    const answers: { selected: any } = await inquirer.prompt([
+      {
+        type: 'autocomplete',
+        name: 'selected',
+        message: message,
+        source: async (answers_so_far: any, input: string) => {
+          return choices.map((p) => ({ name: p.name, value: p }));
+        },
+      },
+    ]);
+    return answers.selected;
+  }
+
+  static async fetchJsonFromGitHub(url: string): Promise<any> {
+    const response = await axios.get(url)
+    .then((res: any) => res.data)
+    .catch((err: any) => {
+      throw new Error(`Failed to fetch ${url}`);
+    });
+    return response;
+  }
+
+  static async fetchYamlFromGitHub(url: string): Promise<ComponentSpec> {
+    const component_spec = await axios.get(url)
+      .then((res: any) => res.data)
+      .catch((err: any) => {
+        throw new Error(`Failed to fetch ${url}`);
+      });
+    return component_spec as ComponentSpec;
+  }
+
+  static async downloadGitHubRepos(selections: Dictionary<Dictionary<any>>, project_dir: string): Promise<void> {
+    // download any selection that has a repository
+    for (const selection of Object.values(selections)) {
+      if (selection.repository) {
+        await execa('git', ['clone', selection.repository, project_dir + '/' + selection.name.toLowerCase()]);
+      }
+    }
+    await new Promise(resolve => setTimeout(resolve, 2000));
+  }
+
+  static async updateServiceEnvironmentDependencies(service_environment: Dictionary<EnvironmentSpecValue>, dep_key: string): Promise<Dictionary<EnvironmentSpecValue>> {
+    const dependencies_regex = new RegExp("'(.*?)'", 'g');
+    for (const [env_key, env_val] of Object.entries(service_environment)) {
+      if (typeof env_val === 'string' && env_val.includes('dependencies')) {
+        const matches = dependencies_regex.exec(env_val);
+        if (matches) {
+          service_environment[env_key] = env_val.replace(matches[0], `'${dep_key.toLowerCase()}'`);
+        }
+      }
+    }
+    return service_environment;
+  }
+
+  static switchDatabase(database_yml: ComponentSpec, backend_yml: ComponentSpec, backend_yml_path: string): void {
+    if (!backend_yml.services || !database_yml.services) {
+      return;
+    }
```

Comment:
Is this still WIP? I tried creating multiple project w/ different database and they still all have the postgre image in them

Reply:
Hmm I'll look into it.